### ISSUE TITLE
Selecting a song to play continues playback of current queue

### ISSUE THREAD

**Issue by flkrz** [lucaspape/catplayer#47]
**Describe the bug**
Searching for a song to play and pressing the song to start playback instead just resumes playback of the current queue

**To Reproduce**
1. Go to search and find any track
2. Click on the track
3. Whatever song was being played last session resumes playback

**Expected behavior**
Clicking the song from.search will immediately start playback of the selected song

**Screenshots**

https://user-images.githubusercontent.com/59532240/109140612-ae4a5f80-7754-11eb-81ef-b695748f7bdd.mp4

**Smartphone (please complete the following information):**
 - Device: SAMSUNG GALAXY NOTE9
 - OS: ANDROID 10
 - Browser N/A
 - Version 1.6.2

**Additional context**
No? App runs great and login is greatly improved since last tested version (quite a while ago)

**Comment by lucaspape** [comment]
Thanks for reporting! I could reproduce this, it is caused by having songs in the priority queue (which has priority)

**Comment by flkrz** [comment]
> Thanks for reporting! I could reproduce this, it is caused by having songs in the priority queue (which has priority)

How can you differentiate between priority queue and normal queue? Want aware this was a feature?

**Comment by lucaspape** [comment]
I differentiate between three things inside the player:

Playlist: the songs that are actually played (to be able to press back and stuff like that)

Queue: songs that should be played, if you press on a song from search it will be added to that queue

Priority queue: long press on a song and press add to queue and it will be added to the priority queue, should also work with playlists and albums

Im not sure if I have a function anywhere to clear this priority queue, the normal queue is cleared every time you press on a song. So thats whats missing here

If you didnt add something to the priority queue and it happens anyway Ill have to look further,
a workaround right now is to force stop the app and then clear the cache in your device settings (that will clear all queues)

**Comment by flkrz** [comment]
Interesting.   
  

  
Why exactly would separate queues be needed? Seems very counterintuitive to have 2 separate ways to setup songs to play. Surely just having one queue for everything being added through either method, with perhaps a user view to manage the queue, would make the most sense? Would also being the app on line with most other music players.   
  

  

  

  
  
  
  
  
>

**Comment by lucaspape** [comment]
Yes making the queues visible was exactly what I intended to do.

It is separate because all "visible" songs are added to the normal queue.
If you are currently playing a playlist and then press "Add to queue" you want that song you just added to queue to end up on top. So I just created a separate queue that always has priority. 
I used a single queue before and it was very bugged for several reasons, by now I could probably unify them again.

I think I didnt want the priority songs to disappear after changing playlists, which is confusing, so Ill have to figure something out.
